### ISSUE TITLE
What is requestAnimationFrame doing?

### ISSUE THREAD

**Issue by tpai** [tpai/tpai.github.io#46]
# TL;DR

The callback of `requestAnimationFrame` will not be blocked by callback flow which created by Web API such as `onscroll` or `setInterval`.

-----

Before knowing rAF(requestAnimationFrame), I will use the code below to create animation in 60 fps(frames per second):

```
let left = 0;

function moveYourBody() {
  document.body.style.marginLeft = `${++left}px`;
  if (left === 60) {
    left = 0;
  }
}

setInterval(moveYourBody, 1000/60);
```

Soon you will find the animation stopped when scrolling the page, because `setInterval` is Web API that defer execution of function, and now callback queue is full of other callback functions which listen to scroll event, so the callback `moveYourBody` is blocking until his turn. (check out [this video](https://www.youtube.com/watch?v=LKGyABwRfgA))

You could imagine `requestAnimationFrame` is a callback function which will notify you when the next  paint be performed. Not like the callback of setInterval needs to wait in the line(queue), the callback of rAF will be called every time browser is ready for paint operation.

Now I know I could write my code like this:

```
let left = 0;

function animate (handle) {
  handle();
  requestAnimationFrame(() => animate(handle));
}

function moveYourBody() {
  document.body.style.marginLeft = `${++left}px`;
  if (left === 60) {
    left = 0;
  }
}

animate(() => moveYourBody());
```
